magento/magento2
# Handle transparncy correctly for watermark

Handle alpha transparency correctly for 24-bit PNG image

### Description
The watermark functionality uses imagecopymerge for copying the
watermark into the image, this function loses the alpha information.

Therefor use imagecopy for 100% opacity and use imagefilter in other
cases.

### Fixed Issues (if relevant)
1. magento/magento2#10661: Opacity png watermark became white box on product images

### Manual testing scenarios
1. Create a 24-bit PNG image with alpha transparency (or use the image in issue in #10661)
2. Configure the image under "Product image watermarks"
3. Leave size empty, set image position to something different than stretch
4. Create a product with a JPEG image
5. See that the watermark is applied with the correct transparency.

### Contribution checklist
